File: kushki-android/src/test/java/com/kushkipagos/android/AurusClientTest.java
package com.kushkipagos.android;

import org.hamcrest.CoreMatchers;
import org.junit.Rule;
import org.junit.Test;
import org.junit.rules.ExpectedException;

import java.io.IOException;

public class AurusClientTest {

    @Rule
    public final ExpectedException expectedException = ExpectedException.none();

    @Test
    public void shouldWrapIOExceptionWithKushkiException() throws Exception {
        AurusClient aurusClient = new AurusClient(TestEnvironment.INVALID, new AurusEncryption());
        expectedException.expect(KushkiException.class);
        expectedException.expectCause(CoreMatchers.<Throwable>instanceOf(IOException.class));
        aurusClient.post("/this-endpoint-does-not-exist", "");
    }
}

File: kushki-android/src/test/java/com/kushkipagos/android/AurusClientTest.kt
package com.kushkipagos.android

import org.hamcrest.CoreMatchers
import org.junit.Rule
import org.junit.Test
import org.junit.rules.ExpectedException

import java.io.IOException

class AurusClientTest {

    @Rule
    @JvmField
    val expectedException: ExpectedException = ExpectedException.none()

    @Test
    fun shouldWrapIOExceptionWithKushkiException() {
        val aurusClient = AurusClient(TestEnvironment.INVALID, AurusEncryption())
        expectedException.expect(KushkiException::class.java)
        expectedException.expectCause(CoreMatchers.instanceOf<Throwable>(IOException::class.java))
        aurusClient.post("/this-endpoint-does-not-exist", "")
    }
}